crates/sui-network/src/state_sync/mod.rs

```diff
@@ -832,42 +855,24 @@ where
         .buffered(checkpoint_header_download_concurrency);
 
     while let Some((maybe_checkpoint, next)) = request_stream.next().await {
+        debug_assert!(current.sequence_number().saturating_add(1) == next);
```

Comment:
`checked_add`? we shouldn't hit u64 max, of course, but if we do i think it would warrant a panic.

Reply:
So saturating_add is used quite prevalently through out this so i'd probably defer making this change to later. Also i'm not sure how i'd feel about needing to sprinkle more `unwraps` everywhere.